server/src/test/java/io/crate/integrationtests/SysShardsTest.java

```diff
@@ -419,4 +423,38 @@ public void testScalarEvaluatesInErrorOnSysShards() throws Exception {
             "select 1/0 from sys.shards"),
                      isSQLError(is("/ by zero"), INTERNAL_ERROR, BAD_REQUEST, 4000));
     }
+
+    @Test
+    public void testSelectFromSysShardsForClosedTables() throws Exception {
+        execute("create table tbl (x int)");
+        execute("insert into tbl values(1)");
+        execute("alter table tbl close");
+        execute("select min_lucene_version, node, path, recovery, retention_leases, seq_no_stats, translog_stats " +
+                "from sys.shards where table_name = 'tbl' and primary = 'true'");
+        assertThat(TestingHelpers.printedTable(response.rows()), not(containsString("NULL")));
+    }
+
+    @UseJdbc(0)
+    @Test
+    public void testSelectFromClosedTableNotAllowed() throws Exception {
+        expectedException.expect(OperationOnInaccessibleRelationException.class);
+        expectedException.expectMessage("The relation \"doc.tbl\" doesn't support or allow READ operations, as it is currently closed.");
```

Comment:
Could you move this just above the `execute` call which raises the exception? It's usually the last call, but it helps avoid mistakes.

 (and maybe convert it to `assertThrows`, the `expectedException` rule is deprecated in the latest junit4 release)